datajoint/blob.py

```diff
@@ -141,7 +141,8 @@ def read_blob(self, n_bytes=None):
                 "F": self.read_recarray,  # numpy array with fields, including recarrays
                 "d": self.read_decimal,  # a decimal
                 "t": self.read_datetime,  # date, time, or datetime
-                "T": self.read_np_datetime,  # np.datetime64
+                "T": self.read_int64_datetime,  # np.datetime64
+                "Z": self.read_np_array_dt64,
```

Comment:
Ideally, we should not use any numpy-specific terminology.

Reply:
The datatype inside arrays is handled in a difference place, inside the `A` datatype.

Reply:
`mxClassID` specifies the codes for different datatypes inside arrays.

Reply:
Unfortunately matlab does not have a c-class for datetimes https://www.mathworks.com/help/matlab/apiref/mxclassid.html, I can add another (key, value) to the mxClassID dict like `NON_mx_DATETIME64, np.dtype("datetime64[us]")` but it wont translate to a matlab api C class. If I add that key then I can add the handling to `read_array` to handle arrays of datetime64. Does this sound like a good way to do it @dimitri-yatsenko?

Reply:
perhaps we can getaway from explicit matching of `mxClassID` and just called it `datatypeID`. Future changes in MATLAB will need to be integrated.

Reply:
Effectively, the serialization protocol has two types of codes: data structures and scalar types.  Scalar types can be used directly in arrays. The current datetime datatype is encoded as the data structure code `t`.  We can add it now as a scalar type too for efficient processing when used in arrays. Numpy's `datetime64` could be adopted.